src/include/lean/lean.h

```diff
@@ -570,26 +623,77 @@ static inline size_t lean_ctor_get_usize(b_lean_obj_arg o, unsigned i) {
     return *((size_t*)(lean_ctor_obj_cptr(o) + i));
 }
 
+/*
+Given a ctor-object, returns a scalar parameter of type `uint8_t` at a given offset. 
+
+The offset does *not* represent an index into the list of parameters of an inductive 
+type constructor, but is instead only a byte offset. To access a specific constructor 
+parameter, construct the correct byte offset as follows. 
+
+All scalar parameters of a ctor-object are represented in a contiguous block following
+the non-scalar parameters. Thus, the base offset required to access a scalar parameter
+of a ctor-object `o` is `lean_ctor_num_objs(o) * sizeof(void*)` (cf. the documentation 
+for `lean_ctor_num_objs` for more details). Then, to locate the desired scalar parameter, 
+add the sizes of  all scalar parameters preceeding the target parameter. For example, 
+let `o` be a value of the following structure in Lean:
+
+```
+structure S where
+    a : String
+    b : Bool
+    c : UInt64
+    e : Array Nat
+    d : UInt8
+```
+
+Then 
+* the base offset is `base_offset := 3 * sizeof(void*)`.
```

Comment:
This should be `2 * sizeof(void*)` I think, since `a` and `e` (why is `e` before `d` in the example?) are the non-scalars.

Reply:
Oh, that is tricky! My eyes bounced right off that.